src/librustc_passes/ast_validation.rs

```diff
@@ -272,7 +272,9 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
                     self.err_handler().span_err(item.span, "inherent impls cannot be negative");
                 }
                 if defaultness == Defaultness::Default {
-                    self.err_handler().span_err(item.span, "inherent impls cannot be default");
+                    self.err_handler()
+                        .struct_span_err(item.span, "inherent impls cannot be default")
+                        .help("maybe a missing `for` keyword?");
```

Comment:
This is creating a `DiagnosticBuilder` but never has `.emit()` called on it, which causes an Internal Compiler Error in test `compile-fail/specialization/defaultimpl/validation.rs`.

Assign the result of the `struct_span_err` to a mutable variable `err` and then `err.help(...)` and `err.emit()`.

Reply:
Oh, okay. Sorry, I'm not very familiar with internals. And there's no rush for feedback, thanks for the warning.